### ISSUE TITLE
Question: How is/will Act be different than Play 1.x?

### ISSUE THREAD

**Issue by megamattron** [actframework/actframework#13]
Hi there, I've been following Act development with interest as I'm a developer that fell in love with Play 1.x  - https://www.playframework.com/documentation/1.2.7/home but didn't like where things headed with Play 2.x.

Do any of you have experience with Play 1? Any thoughts or ideas on how it will be similar or different?

Thanks!

**Comment by greenlaw110** [comment]
@megamattron Yes I am also a lover and heavy user of play 1.x. I've contributed quite a few modules to Play and five of them are in the registered module list:

* [play-moprhia](https://www.playframework.com/modules/morphia)
* [play-rythm](https://www.playframework.com/modules/rythm)
* [play-excel](https://www.playframework.com/modules/excel)
* [greenscript](https://www.playframework.com/modules/greenscript)
* [play-menu](https://www.playframework.com/modules/menu)

The thing with play 2.x is a little bit shift away from my expectation. I have been struggling with Sprint MVC and Jersey for a few projects after play 2.x announced. However I decide to rule out my own MVC stack in the end and that's where Act come from.

Checkout http://software-lgl.blogspot.com.au/2016/03/introduce-to-actframework-new-java-mvc.html to get some idea on where Act is different from Play

**Comment by megamattron** [comment]
Thank you - interesting link. I have a couple questions for you after reading it:

1. Is Act a share-nothing framework like Play?
1. Did you ever consider starting from the Play 1 source and continuing from there? Was there something  in Play 1 that made you want to start over? It seems like you are re-implementing the main features of Play with a couple substitutions like EBean for JPA, etc. I'm wondering what caused the decision to start from scratch.
1. How much code generation is used? I have a pretty large Play 1 project and the memory usage was getting fairly significant, especially PermSize and stack size. That was one of the few things I didn't like about Play 1.

Thanks!

**Comment by greenlaw110** [comment]
> 1. Is Act a share-nothing framework like Play?

Yes

> Did you ever consider starting from the Play 1 source and continuing from there? 

Once upon a time, Someone and I have considered [this](https://github.com/greenlaw110/yalp). But we failed on that unfortunately

>Was there something in Play 1 that made you want to start over? 

See http://software-lgl.blogspot.com.au/2012/12/thinking-about-creating-new-java-web.html

>It seems like you are re-implementing the main features of Play with a couple substitutions like EBean for JPA, etc. I'm wondering what caused the decision to start from scratch.

EBean is just one tiny piece of things in my consideration. Other things I've been thinking about Act when I started the project:

* It should be maven standard project and easy to integrate with other libraries

* Bytecode enhancement is cool, but I want to re-write it with ASM for better performance. Also I don't like the property enhancement in Play as it is little bit against the instinct of normal Java programmer. So I removed that piece in Act

* I think the controller pattern should be more flexible and there is no reason to restrict action method be `public static void`

* I prefer to support JSR330 standard for dependency injection

* Play's Job feature is quite nice, but there is room to improve

* There is a missing feature event handling in Play

* The routing mechanism of Play is not very cool and it could be very slow if you got a huge routing table

* Plugin mechanism is a bit silly, there is no reason to implement `modelFactory` and `afterFixtureLoad` if I just want to plugin a different templating solution

* I need the framework support multi-DB and even multiple heterogeneous DB so that my application can talk to mongdb and mysql at the same time

>How much code generation is used? I have a pretty large Play 1 project and the memory usage was getting fairly significant, especially PermSize and stack size. That was one of the few things I didn't like about Play 1.

Bytecode enhancements are there for

* [Controller action methods](https://github.com/actframework/act-doc/blob/master/en/controller.md)
* [Database model classes](https://github.com/actframework/act-doc/blob/master/en/model.md) in a much lighter way in comparing to Play 1
* [Console commander methods](https://github.com/actframework/act-doc/blob/master/en/cli.md)
* [Job methods](https://github.com/actframework/act-doc/blob/master/en/jobs.md)
* [Mailer methods](https://github.com/actframework/act-doc/blob/master/en/email.md)
* [Event handler](https://github.com/actframework/act-doc/blob/master/en/event.md)

However all those bytecode enhancements are very lightweight and no big chunk of code has been added to your original class bytecode. 

Since actframework does not flavor static methods for controller, I think at least perm size issue should be alleviated if your project switch to Act. But anyway you won't have the perm size issue in Java 8
